src/views/DatasetManager/components/UpdateDatasetButton/UpdateDatasetButton.js

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+import { React, useCallback, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+import EditIcon from '@mui/icons-material/Edit';
+import { IconButton } from '@mui/material';
+import { FadingTooltip, PermissionsGate } from '@cosmotech/ui';
+import { INGESTION_STATUS } from '../../../../services/config/ApiConstants';
+import { ACL_PERMISSIONS } from '../../../../services/config/accessControl';
+import { useGetAllRunners } from '../../../../state/hooks/RunnerHooks';
+import { UpdateDatasetDialog } from './components';
+
+export const UpdateDatasetButton = ({ dataset }) => {
+  const { t } = useTranslation();
+  const runners = useGetAllRunners();
+  const selectedRunner = runners.find((runner) => runner.id === dataset.source.name);
+
+  const isDisabled =
+    (dataset && dataset.ingestionStatus === INGESTION_STATUS.PENDING) ||
+    !selectedRunner ||
+    Object.keys(selectedRunner ?? {})?.length === 0;
```

Comment:
`!selectedRunner` seems to be redundant with `Object.keys(selectedRunner ?? {})?.length === 0`, I think we could keep the second one and remove the part `!selectedRunner` :slightly_smiling_face: 
Also, `dataset && dataset.ingestionStatus === INGESTION_STATUS.PENDING` could probably be simplified into `dataset?.ingestionStatus === INGESTION_STATUS.PENDING`